iceoryx_utils/include/iceoryx_utils/posix_wrapper/thread.hpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef IOX_UTILS_POSIX_WRAPPER_PTHREAD_HPP
+#define IOX_UTILS_POSIX_WRAPPER_PTHREAD_HPP
+
+#include "iceoryx_utils/cxx/expected.hpp"
+#include "iceoryx_utils/cxx/smart_c.hpp"
+#include "iceoryx_utils/cxx/string.hpp"
+#include "iceoryx_utils/platform/pthread.hpp"
+
+namespace iox
+{
+namespace posix
+{
+enum class ThreadErrorType : int32_t
```

Comment:
This `ThreadErrorType` is no where used and could be removed.

Reply:
Done